core/src/main/java/org/mskcc/cbio/portal/scripts/ImportMutationalSignatureData.java

```diff
@@ -0,0 +1,164 @@
+            String description = "Import mutational signature data files.";
+
+            OptionParser parser = new OptionParser();
+            OptionSpec<String> data = parser.accepts("data", "Mutational Signature data file")
+                .withRequiredArg().ofType(String.class);
+            OptionSpec<Void> help = parser.accepts( "help", "print this help info" );
+            parser.accepts( "mutational-signatures", "mutational signatures file" ).withRequiredArg().describedAs( "mutational_signature.txt" ).ofType( String.class );
+            parser.accepts("update-info", "Only update the info of mutational signatures");
+
+            OptionSet options = null;
+            try {
+                options = parser.parse(args);
+            }
+            catch (Exception ex) {
+                throw new UsageException(
+                    progName, description, parser,
+                    ex.getMessage());
+            }
+
+            if( options.has( help ) ){
+                throw new UsageException(progName, description, parser);
+            }
+            
+            // Check options
+            boolean updateInfo = options.has("update-info");
+
+            ProgressMonitor.setConsoleMode(true);
+
+            File mutationalSignatureFile;
+            int numLines;
+            if(options.has("data")) {
+                mutationalSignatureFile = new File((String) options.valueOf("data"));
+
+                System.out.println("Reading gene data from:  " + mutationalSignatureFile.getAbsolutePath());
+                numLines = FileUtil.getNumLines(mutationalSignatureFile);
+                System.out.println(" --> total number of lines:  " + numLines);
+                ProgressMonitor.setMaxValue(numLines);
+                ImportMutationalSignatureData.importData(mutationalSignatureFile, updateInfo);
+            }
+            
+        } catch (RuntimeException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    /**
+     * Adds the mutational signatures parsed from the file into the Database.
+     *
+     * @param mutationalSignatureFile File with mutational signature information
+     * @throws IOException
+     * @throws DaoException
+     */
+    
+    public static void importData (File mutationalSignatureFile, boolean updateInfo) throws IOException, DaoException{
+        try (FileReader reader = new FileReader(mutationalSignatureFile)) {
+            BufferedReader buf = new BufferedReader(reader);
+            String line;
+            while ((line = buf.readLine()) != null){
+                ProgressMonitor.incrementCurValue();
+                ConsoleUtil.showProgress();
+                
+                String[] parts = line.split("\t");
+
+                //assume mutational signature id and description are first two columns in file
+                MutationalSignature mutationalSignature = new MutationalSignature();
```

Comment:
I would add some checks to parts to make sure fields are not empty before creating a new MutationalSignature() and setting its fields.